File: generators/goclient/testdata/todo_client.go
// Client is the API client.
type Client struct {
  // URL is the required API endpoint address.
  URL string

  // AuthToken is an optional authentication token.
  AuthToken string

  // HTTPClient is the client used for making requests, defaulting to http.DefaultClient.
  HTTPClient *http.Client
}

// AddItem adds an item to the list.
func (c *Client) AddItem(in AddItemInput) error {
  return call(c.HTTPClient, c.AuthToken, c.URL, "add_item", in, nil)
}

// GetItems returns all items in the list.
func (c *Client) GetItems() (*GetItemsOutput, error) {
  var out GetItemsOutput
  return &out, call(c.HTTPClient, c.AuthToken, c.URL, "get_items", nil, &out)
}

// RemoveItem removes an item from the to-do list.
func (c *Client) RemoveItem(in RemoveItemInput) (*RemoveItemOutput, error) {
  var out RemoveItemOutput
  return &out, call(c.HTTPClient, c.AuthToken, c.URL, "remove_item", in, &out)
}


// Error is an error returned by the client.
type Error struct {
	Status     string
	StatusCode int
	Type       string
	Message    string
}

// Error implementation.
func (e Error) Error() string {
	if e.Type == "" {
		return fmt.Sprintf("%s: %d", e.Status, e.StatusCode)
	}
	return fmt.Sprintf("%s: %s", e.Type, e.Message)
}

// call implementation.
func call(client *http.Client, authToken, endpoint, method string, in, out interface{}) error {
	var body io.Reader

	// default client
	if client == nil {
		client = http.DefaultClient
	}

	// input params
	if in != nil {
		var buf bytes.Buffer
		err := json.NewEncoder(&buf).Encode(in)
		if err != nil {
			return fmt.Errorf("encoding: %w", err)
		}
		body = &buf
	}

	// POST request
	req, err := http.NewRequest("POST", endpoint+"/"+method, body)
	if err != nil {
		return err
	}
	req.Header.Set("Content-Type", "application/json")

	// auth token
	if authToken != "" {
		req.Header.Set("Authorization", "Bearer "+authToken)
	}

	// response
	res, err := client.Do(req)
	if err != nil {
		return err
	}
	defer res.Body.Close()

	// error
	if res.StatusCode >= 300 {
		var e Error
		if res.Header.Get("Content-Type") == "application/json" {
			if err := json.NewDecoder(res.Body).Decode(&e); err != nil {
				return err
			}
		}
		e.Status = http.StatusText(res.StatusCode)
		e.StatusCode = res.StatusCode
		return e
	}

	// output params
	if out != nil {
		err = json.NewDecoder(res.Body).Decode(out)
		if err != nil {
			return err
		}
	}

	return nil
}

File: generators/dotnetclient/testdata/todo_client.cs
using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Threading.Tasks;
using Newtonsoft.Json;

namespace ApexLogs
{
	public class Client
	{
		class ApexLogsException : Exception
		{
			public ApexLogsException(int status) : base($"{status} response") 
			{ }

			public ApexLogsException(int status, string type, string message) 
				: base($"{status} response: ${type}: {message}") 
			{ }
		}

		private readonly string _url;
		private readonly string _authToken;
		private readonly HttpClient _httpClient;

		public Client(HttpClient httpClient, string url, string authToken)
		{
			_httpClient = httpClient;
			_url = url;
			_authToken = authToken;
		}

		/// adds an item to the list.
		public async Task AddItem(AddItemInput parameter)
		{
			await Call("add_item", parameter);
		}

		/// returns all items in the list.
		public async Task<GetItemsOutput> GetItems()
		{
			var res = await Call("get_items");
			var output = JsonConvert.DeserializeObject<GetItemsOutput>(res);
			return output;
		}

		/// removes an item from the to-do list.
		public async Task<RemoveItemOutput> RemoveItem(RemoveItemInput parameter)
		{
			var res = await Call("remove_item", parameter);
			var output = JsonConvert.DeserializeObject<RemoveItemOutput>(res);
			return output;
		}

		public async Task<string> Call(string method, object parameters = null)
		{
			var url = $"{_url}/{method}";
			var message = new HttpRequestMessage
			{
				Method = HttpMethod.Post,
				RequestUri = new Uri(url)
			};
			message.Headers.Add("Content-Type", "application/json");
			if (!string.IsNullOrWhiteSpace(_authToken))
				message.Headers.Add("Authorization", $"Bearer {_authToken}");

			if (parameters != null)
				message.Content = new StringContent(JsonConvert.SerializeObject(parameters));

			var response = await _httpClient.SendAsync(message);
			var statusCode = (int) response.StatusCode;
			var content = await response.Content.ReadAsStringAsync();

			if (statusCode < 300) return content;

			var body = JsonConvert.DeserializeObject<Dictionary<string, string>>(content)
				?? throw new ApexLogsException(statusCode);

			throw new ApexLogsException(statusCode, body["type"], body["message"]);
		}
	}
}
